### PR TITLE
feat: Self Attested Claims (VCs) - id_token & custom scope

### PR DESCRIPTION
**Pull request by mishasizov-SK**

### CODE REVIEW

File: component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go [hunk 1]
```diff
@@ -602,6 +603,21 @@ func (f *Flow) createIDToken(
 	return idTokenJSON, nil
 }
 
+func (f *Flow) extractAdditionalClaims(requestObjectScope string) map[string]Claims {
```
Comment (mishasizov-SK) [inline component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go:606]: As discussed, for now wallet returns random claims and does not considers custom scope.
CC: @sudeshrshetty  @vkubiv

File: pkg/service/oidc4vp/oidc4vp_service.go [hunk 2]
```diff
@@ -387,6 +397,16 @@ func (s *Service) VerifyOIDCVerifiablePresentation(ctx context.Context, txID TxI
 			fmt.Errorf("invalid nonce"))
 	}
 
+	// If custom scope was requested, but no relevant additional claims were supplied by Holder.
+	noAdditionalClaims := tx.CustomScope != "" && len(authResponse.CustomScopeClaims[tx.CustomScope]) == 0
+	// If custom scope was not requested, but additional claims were supplied by Holder.
+	unexpectedAdditionalClaims := tx.CustomScope == "" && len(authResponse.CustomScopeClaims) != 0
+
```
Comment (mishasizov-SK) [inline pkg/service/oidc4vp/oidc4vp_service.go:404]: Verifier checks if there additional claims supplied for given scope.

File: component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go (hunk 1 above)
Comment (vkubiv) [reply component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go:606]: @mishasizov-SK  What do you mean by random? Any?

Comment (sudeshrshetty) [reply component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go:606]: I am not sure what you mean, you can pre-configure wallet CLI with pre-defined "scope:claim", and ask the exact scope while testing.

Comment (mishasizov-SK) [reply component/wallet-cli/pkg/oidc4vp/oidc4vp_flow.go]: @sudeshrshetty updated

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**